src/validators/url.rs

```diff
@@ -163,27 +181,27 @@ impl<'py> IntoPyObject<'py> for EitherUrl<'py> {
 
     fn into_pyobject(self, py: Python<'py>) -> PyResult<Self::Output> {
         match self {
-            EitherUrl::Py(py_url) => Ok(py_url),
-            EitherUrl::Rust(rust_url) => Bound::new(py, PyUrl::new(rust_url)),
+            EitherUrl::Bound(py_url) => Ok(py_url),
+            EitherUrl::Owned(py_url) => Bound::new(py, py_url),
         }
     }
 }
 
 impl CopyFromPyUrl for EitherUrl<'_> {
```

Comment:
Seeing this again, this looks like it probably should have been `AsRef<Url>` and `AsMut<Url>` implementations.